tests/testthat/test-settings.R

```diff
@@ -152,3 +152,94 @@ test_that("it has a smart default for encodings", {
   lintr:::read_settings(pkg_file)
   expect_identical(settings$encoding, "ISO8859-1")
 })
+
+test_that("validate_config_file() detects improperly-formed settings", {
+  .lintr <- withr::local_tempfile()
+  withr::local_options(lintr.linter_file = .lintr)
+  withr::local_dir(withr::local_tempdir())
+
+  writeLines("asdf: 1", .lintr)
+  expect_warning(lint_dir(), "Found unused settings in config", fixed = TRUE)
+
+  writeLines("a=1", "aaa.R")
+  writeLines(c('exclusions: list("aaa.R")', "asdf: 1"), .lintr)
+  expect_warning(lint_dir(), "Found unused settings in config", fixed = TRUE)
+
+  writeLines("encoding: FALSE", .lintr)
+  expect_error(lint_dir(), "Setting 'encoding' should be a character string, not 'FALSE'", fixed = TRUE)
+
+  writeLines("encoding: NA_character_", .lintr)
+  expect_error(lint_dir(), "Setting 'encoding' should be a character string, not 'NA'", fixed = TRUE)
+
+  writeLines("encoding: letters", .lintr)
+  expect_error(lint_dir(), "Setting 'encoding' should be a character string, not '[a-z, ]+'")
```

Comment:
What's up with that expectation? `rex()` clash?

Reply:
Hmm I think `letters` is obscuring this. Better to use a simple literal instead for clarity.